code/modules/mob/living/carbon/xenomorph/life.dm

```diff
@@ -338,7 +338,7 @@ Make sure their actual health updates immediately.*/
 
 	if(caste)
 		if(caste.innate_healing || check_weeds_for_healing())
-			if(!hive) return // can't heal if you have no hive, sorry bud
+			if(!hive || on_fire) return // can't heal if you have no hive, sorry bud
```

Comment:
this will completely cut off the rest of the proc from running, which does not seem ideal - you'll prevent regular things such as ability icons updating, actions running the `life_tick`, as well as preventing pheros from being cancelled if you are out of plasma.

You're going to want to remove this early return, it makes little sense here anyways. Just add this into the above checks